### PR TITLE
Use framework.ExpectEqual() in unit test

### PR DESCRIPTION
**Pull request by clarklee92**
Signed-off-by: clarklee92 <clarklee1992@hotmail.com>

**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Replace `gomega.Expect().To(gomega.Equal(), ...)` to `framework. ExpectEqual()` in
`test/e2e/framework/log_test.go` and `test/e2e/framework/timer/timer_test.go`

**Which issue(s) this PR fixes**:

Ref #79686

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/log_test.go [hunk 1]
```diff
@@ -117,16 +117,16 @@ func TestFailureOutput(t *testing.T) {
 		},
 	}
 	// Compare individual fields. Comparing the slices leads to unreadable error output when there is any mismatch.
-	g.Expect(len(actual)).To(gomega.Equal(len(expected)), "%d entries in %v", len(expected), actual)
```
Comment (oomichi) [inline test/e2e/framework/log_test.go]: need to remove line 86 due to `b/f/w/test/e2e/framework/log_test.go:86:2: g declared and not used`

Comment (clarklee92) [reply test/e2e/framework/log_test.go]: Thanks, already updated.

**Review by alejandrox1 — COMMENTED** [review]
Thank you for working on this @clarklee92 
/lgtm
/kind cleanup
/priority important-soon
/milestone v1.17

**Pull request merged**